cmd/oras/push.go

```diff
@@ -132,7 +137,13 @@ func packManifest(ctx context.Context, store *file.Store, annotations map[string
 	packOpts.ConfigAnnotations = annotations[annotationConfig]
 	packOpts.ManifestAnnotations = annotations[annotationManifest]
 
+	if opts.artifactTpye != "" {
+		packOpts.ConfigMediaType = opts.artifactTpye
+	}
 	if opts.manifestConfigRef != "" {
+		if opts.artifactTpye != "" {
+			fmt.Println("--manifest-config will override --artifact-type")
```

Comment:
According to the [discuz](https://cloud-native.slack.com/archives/CJ1KHJM5Z/p1658886080869469), we should exit with error so user will not abuse the overriding.